src/Surfnet/AzureMfa/Infrastructure/Validator/Constraint/EmailDomainInConfigurationValidator.php

```diff
@@ -0,0 +1,65 @@
+<?php declare(strict_types=1);
+
+/**
+ * Copyright 2019 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\AzureMfa\Infrastructure\Validator\Constraint;
+
+use Surfnet\AzureMfa\Application\Institution\Service\EmailDomainMatchingService;
+use Surfnet\AzureMfa\Domain\EmailAddress;
+use Surfnet\AzureMfa\Domain\Exception\InvalidEmailAddressException;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+class EmailDomainInConfigurationValidator extends ConstraintValidator
+{
+    /**
+     * @var EmailDomainMatchingService
+     */
+    private $matchingService;
+
+    public function __construct(EmailDomainMatchingService $service)
+    {
+        $this->matchingService = $service;
+    }
+
+    public function validate($value, Constraint $constraint)
```

Comment:
Is a type hint not required?

Reply:
The declaration must be compatible with ConstraintValidatorInterface->validate(value, constraint: \Symfony\Component\Validator\Constraint). Best I could do was add a mixed typehint in a param annotation.